pkg/kapp/crdupgradesafety/change_validator.go

```diff
@@ -113,6 +113,98 @@ func RequiredFieldChangeValidation(diff FieldDiff) (bool, error) {
 	return handled(), nil
 }
 
+func MinimumChangeValidation(diff FieldDiff) (bool, error) {
+	handled := func() bool {
+		diff.Old.Minimum = nil
+		diff.New.Minimum = nil
+		return reflect.DeepEqual(diff.Old, diff.New)
+	}
+
+	switch {
+	case diff.Old.Minimum == nil && diff.New.Minimum != nil:
+		m := *diff.New.Minimum
+		return handled(), fmt.Errorf("minimum constraint added when one did not exist previously: %+v", m)
```

Comment:
Can `*diff.New.Minimum` be referenced here directly  without having to create `m`:

```suggestion
		return handled(), fmt.Errorf("minimum constraint added when one did not exist previously: %+v", *diff.New.Minimum)
```

Reply:
That is what I originally had before writing the e2e tests, but noticed that this resulted in the output being `<nil>` due to the execution order of the return statement. the `handled()` function resets the values to `nil`. This is so that we can ensure that other checks only have to change their field and check for equality and each check can assume every other check is doing the same.

Reply:
Capturing the value in a new variable before the return statement is executed allows us to have the more explicit and clear error message.

Reply:
Got it. Marking all the comments as resolved.

Reply:
Unable to mark these comments as resolved. Feel free to do so on your end if you are able to.